l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -547,45 +575,27 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         uint16 _l2TxNumberInBatch,
         bytes calldata _message,
         bytes32[] calldata _merkleProof
-    ) internal nonReentrant whenNotPaused returns (address l1Receiver, address l1Token, uint256 amount) {
+    ) internal nonReentrant whenNotPaused returns (bytes32 assetId, uint256 amount) {
+        // ToDo: unused function param
         require(!isWithdrawalFinalized[_chainId][_l2BatchNumber][_l2MessageIndex], "Withdrawal is already finalized");
         isWithdrawalFinalized[_chainId][_l2BatchNumber][_l2MessageIndex] = true;
 
         // Handling special case for withdrawal from zkSync Era initiated before Shared Bridge.
-        if (_isEraLegacyEthWithdrawal(_chainId, _l2BatchNumber)) {
-            // Checks that the withdrawal wasn't finalized already.
-            bool alreadyFinalized = IGetters(ERA_DIAMOND_PROXY).isEthWithdrawalFinalized(
-                _l2BatchNumber,
-                _l2MessageIndex
-            );
-            require(!alreadyFinalized, "Withdrawal is already finalized 2");
-        }
-
-        MessageParams memory messageParams = MessageParams({
-            l2BatchNumber: _l2BatchNumber,
-            l2MessageIndex: _l2MessageIndex,
-            l2TxNumberInBatch: _l2TxNumberInBatch
-        });
-        (l1Receiver, l1Token, amount) = _checkWithdrawal(_chainId, messageParams, _message, _merkleProof);
-
-        if (!hyperbridgingEnabled[_chainId]) {
-            // Check that the chain has sufficient balance
-            require(chainBalance[_chainId][l1Token] >= amount, "ShB not enough funds 2"); // not enough funds
-            chainBalance[_chainId][l1Token] -= amount;
+        require(!_isEraLegacyEthWithdrawal(_chainId, _l2BatchNumber), "ShB: legacy eth withdrawal");
```

Comment:
I don't see a method to finalize an old withdrawal. Does it mean that if there was an old withdrawal, users wont be able to finalize it? 

I personally believe that it may be acceptable to have this logic to remove legacy, but I would wait a bit longer before removing this method... In any case, you must consult with the product team before doing it